exe.c

```diff
@@ -1824,6 +1824,32 @@ void codegen_statement(ast node) {
     cgc_fs = save_fs;
     cgc_locals = save_locals;
 
+  } else if (op == DO_KW) {
+
+    lbl1 = alloc_label(); // do statement start
+    lbl2 = alloc_label(); // break point
+
+    save_fs = cgc_fs;
+    save_locals = cgc_locals;
+
+    cgc_add_enclosing_loop(cgc_fs, lbl2, lbl1);
+    def_label(lbl1);
+
+    if (get_child(node, 0)) {
+      codegen_statement(get_child(node, 0));
```

Comment:
`codegen_statement` returns early if it's argument is null :)

Reply:
Cool!